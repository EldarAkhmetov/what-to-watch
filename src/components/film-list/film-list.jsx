import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import FilmCard from '../film-card/film-card.jsx';

class FilmList extends PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      activeCardId: null
    };
  }

  render() {
    const {films, currentGenre} = this.props;
    return (
      <div className="catalog__movies-list">
        {films
          .filter(({genre}) => {
            if (currentGenre === `all`) {
              return true;
            }
            return genre === currentGenre;
          })
          .map((it, ind) => {
            return <FilmCard
              key={`card-${ind}-${it.id}`}
              film={it}
              onCardHover={this._cardHoverHandler.bind(this)}
            />;
          })}
      </div>
    );
  }

  _cardHoverHandler(cardId) {
    this.setState({
      activeCardId: cardId
    });
  }
}

FilmList.propTypes = {
  films: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number,
    title: PropTypes.string,
    image: PropTypes.string
  })),
  currentGenre: PropTypes.string
};

const mapStateToProps = (state, ownProps) => Object.assign(
    {},
    ownProps,
    {currentGenre: state.currentGenre}
);

export {FilmList};

export default connect(mapStateToProps)(FilmList);
