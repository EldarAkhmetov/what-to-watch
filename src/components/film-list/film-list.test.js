import React from 'react';
import renderer from 'react-test-renderer';
import {FilmList} from './film-list.jsx';

it(`Film list correctly renders after relaunch`, () => {
  const mock = [
    {
      id: 1,
      title: `12`,
      image: ``,
      genre: ``
    }
  ];

  const tree = renderer
    .create(<FilmList
      films={mock}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
