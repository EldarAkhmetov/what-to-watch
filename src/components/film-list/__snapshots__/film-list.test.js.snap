// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Film list correctly renders after relaunch 1`] = `
<div
  className="catalog__movies-list"
/>
`;
